cpp/bench/ann/CMakeLists.txt

```
@@ -213,20 +242,48 @@ if(RAFT_ANN_BENCH_USE_RAFT_CAGRA)
   )
 endif()
 
+set(RAFT_FAISS_TARGETS faiss::faiss)
+if(CXX_AVX_FOUND)
+  list(APPEND RAFT_FAISS_TARGETS faiss::faiss_avx2)
+endif()
```

This way it will pick up non AVX symbols from `faiss::faiss`, resulting in slower binary. The following worked for me:
```suggestion
if(CXX_AVX_FOUND)
  set(RAFT_FAISS_TARGETS faiss::faiss_avx2)
else()
  set(RAFT_FAISS_TARGETS faiss::faiss)
endif()
```

---

No need to check a variable, just query if the avx2 target exists.
```cmake
set(RAFT_FAISS_TARGETS faiss::faiss)
if(TARGET faiss::faiss_avx2)
  set(RAFT_FAISS_TARGETS faiss::faiss_avx2)
endif()
```